homework-submission/week2-homework/src/App.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import "./App.css";
+import TodoItem from "./components/TodoItem";
+import List from "./components/List";
+import Checkbox from "./components/Checkbox";
+import todos from "./source/todos.json";
+import Header from "./components/Header";
+import image from "./todo.jpg";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(id) {
+    let todos = this.state.todos.slice();
+    let foundTodo = todos.find(todo => id === todo.id);
+    foundTodo.done = !foundTodo.done;
+    this.setState({ todos: todos });
+  }
+
+  render() {
+    let todoItems = this.state.todos.map((todo, index) => (
+      <TodoItem key={index} {...todo}>
+        <Checkbox
+          key={index}
+          done={todo.done}
+          onClick={() => {
+            this.handleClick(todo.id);
```

Comment:
This is the function that toggles todos' done state. I pass it here as Checkbox component props. And I assign it to the new function to prevent an infinite loop because I was getting this error `Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.`

Reply:
Aha I see I am sorry, My bad 🙃